tests/testthat/test_xyz.r

```diff
@@ -0,0 +1,92 @@
+
+## create a set of geometries
+## p, mp, ml, pol, mpol 
+## pz, mpz, mlz, polz, mpolz
+
+p <- st_point(c(0, 0))
+pz <- st_point(c(0, 0, 0))
+
+mp <- st_multipoint(rbind(p, p * 2))
+mpz <- st_multipoint(rbind(pz, pz * 2))
+
+l1 <- cbind(0:1, 0:1)
+l1z <- cbind(0:1, 0:1, 0:1)
+
+l <- st_linestring(l1)
+lz <- st_linestring(l1z)
+
+ml <- st_multilinestring(replicate(2, l1, simplify = FALSE))
+mlz <- st_multilinestring(replicate(2, l1z, simplify = FALSE))
+
+p1 <- rbind(c(0,0), c(1,0), c(3,2), c(2,4), c(1,4), c(0,0))
+p2 <- rbind(c(1,1), c(1,2), c(2,2), c(1,1))
+pol1 <- list(p1,p2)
+pol1z <- list(cbind(p1, 0), cbind(p2, 0))
+
+pol <- st_polygon(pol1)
+polz <- st_polygon(pol1z)
+
+mpol <- st_multipolygon(list(pol1, pol1))
+mpolz <- st_multipolygon(list(pol1z, pol1z))
+
+## worker functions to apply transformations
+## do we need to test inverse versions of these as well? vertical? (see below)
+do_planar <- function(x) st_transform(st_sfc(x, crs = 4326), crs = "+proj=laea")
+do_geo <- function(x) st_transform(st_sfc(x, crs = 4326), crs = "+proj=geocent")
```

Comment:
Ok that's good, it clarifies about what I need to do to go further. As it is I'm happy to ignore this PR since I just won't be able to get to vertical datums, or to tease out what I found here, for a while. I don't agree about the "vertical datums only" part but I can't support that with any great weight now. As long as it's not precluded then I don't have a problem.